bot/cogs/help.py

```diff
@@ -70,499 +68,307 @@ class HelpSession:
-        if not self.is_first_page:
-            await self.update_page(0)
-
-    async def do_back(self) -> None:
-        """Event that is called when the user requests the previous page."""
-        if not self.is_first_page:
-            await self.update_page(self._current_page-1)
-
-    async def do_next(self) -> None:
-        """Event that is called when the user requests the next page."""
-        if not self.is_last_page:
-            await self.update_page(self._current_page+1)
-
-    async def do_end(self) -> None:
-        """Event that is called when the user requests the last page."""
-        if not self.is_last_page:
-            await self.update_page(len(self._pages)-1)
-
-    async def do_stop(self) -> None:
-        """Event that is called when the user requests to stop the help session."""
-        await self.message.delete()
-
-
-class Help(DiscordCog):
-    """Custom Embed Pagination Help feature."""
+        lines = []
 
-    @commands.command('help')
-    @redirect_output(destination_channel=Channels.bot, bypass_roles=STAFF_ROLES)
-    async def new_help(self, ctx: Context, *commands) -> None:
-        """Shows Command Help."""
-        try:
-            await HelpSession.start(ctx, *commands)
-        except HelpQueryNotFound as error:
-            embed = Embed()
-            embed.colour = Colour.red()
-            embed.title = str(error)
+        for cog_or_category, _commands in itertools.groupby(filter_commands, key=self._category_key):
+            sorted_commands = sorted(_commands, key=lambda c: c.name)
 
-            if error.possible_matches:
-                matches = '\n'.join(error.possible_matches.keys())
-                embed.description = f'**Did you mean:**\n`{matches}`'
+            if len(sorted_commands) == 0:
+                continue
 
-            await ctx.send(embed=embed)
+            fmt = [
+                f"`{PREFIX}{c.qualified_name}{f' {c.signature}' if c.signature else ''}`"
+                f"\n*{c.short_doc or 'No details provided.'}*" for c in sorted_commands
+            ]
 
+            # we can't embed a '\n'.join() inside an f-string so this is a bit of a compromise
+            def get_fmt(i: int) -> str:
+                """Get a formatted version of commands for an index."""
+                return "\n".join(fmt[i:i+COMMANDS_PER_PAGE])
 
-def unload(bot: Bot) -> None:
-    """
-    Reinstates the original help command.
-
-    This is run if the cog raises an exception on load, or if the extension is unloaded.
-    """
-    bot.remove_command('help')
-    bot.add_command(bot._old_help)
-
+            # this is a bit yuck because moderation category has 8 commands which needs to be split over 2 pages.
+            # pretty much it only splits that category, but also gives the number of commands it's adding to
+            # the pages every iteration so we can easily use this below rather than trying to split the string.
```

Comment:
This is a bit confusing since currently the Moderation category is small enough to not require a split (has 8 commands and `COMMANDS_PER_PAGE` is also 8). Here's my attempt at a more generic version:

> Split cogs or categories which have too many commands to fit in one page. The length of commands is included for later use when aggregating into pages for the paginator.